### DISCUSSION TITLE
Is Powershell Math broken?

### DISCUSSION THREAD

**Discussion by LawrenceTattrie** [PowerShell/PowerShell#18199] [Q&A] [answered]
Is Powershell math broken? I understand that powershell uses degrees for sin and cos.
[math]::sin(90.0)
0.893996663600558
This does not even give a single digit of accuracy.
[double][math]::sin(90)
0.893996663600558
The "scientific calculator" which comes with windows and python agree that 
>>> math.sin(math.radians(90.0))
1.0
So should sin, cos, tan be removed from the language if powershell is not giving the correct result?

This [](https://4sysops.com/archives/do-the-math-with-powershell/) example, search for sin(90), also shows the error.
I understand the difference between floating point and decimal math. I just don't understand  what is going on here that is so bad.

**Comment by LawrenceTattrie** [comment]
Get-Host | Select-Object Version

Version
-------
7.1.5

**Comment by LawrenceTattrie** [comment, marked as answer]
Apparently some of the docs are wrong and the default is radians.
$deg=90
$rad = $deg * ([Math]::PI/180)
[double][math]::sin($rad)
1

**Reply by jborean93** [reply]
Just an FYI, this isn't PowerShell but rather dotnet itself. The syntax `[Type]::Method($args)` is calling the static member on the type, for example `[Math]::PI` is getting the [PI field on the Math class](https://learn.microsoft.com/en-us/dotnet/api/system.math.pi?view=net-7.0) and `[Math]::Sin` is calling the [Sin method on the Math class](https://learn.microsoft.com/en-us/dotnet/api/system.math.sin?view=net-7.0). The docs there usually have more info, for example the `Sin` method states for the argument

> An angle, measured in radians.

When in doubt, best to check the documentation.
